string data_dir
---
bool success
string message